subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/transport/LegacyTransport.kt

```diff
@@ -0,0 +1,31 @@
+package com.avito.android.test.report.transport
+
+import com.avito.android.test.report.ReportState
+import com.avito.android.test.report.model.TestMetadata
+import com.avito.filestorage.FutureValue
+import com.avito.filestorage.RemoteStorage
+
+/**
+ * Save report to extrnal storage
+ * Send all data to remote storage
+ *
+ * Legacy way; see [ExternalStorageTransport]
+ */
+class LegacyTransport(
```

Comment:
Can it be internal?

Reply:
nope, used in inhouse

Reply:
I meant to hide concrete implementations behind a factory

Reply:
yep, i think its possible; ill try

Reply:
done